lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
Thank you for digging out that issue, unfortunately, `/\s+/` instead of `/   /` will not work, because `-----BEGIN CERTIFICATE-----` contains space. How about `/[\t]+/` as nginx is [adding \t](https://github.com/nginx/nginx/blob/master/src/event/ngx_event_openssl.c#L3280) to cert ?